core/io/multiplayer_api.cpp

```diff
@@ -139,6 +139,9 @@ void MultiplayerAPI::set_network_peer(const Ref<NetworkedMultiplayerPeer> &p_pee
 
 	if (p_peer == network_peer) return; // Nothing to do
 
+	ERR_FAIL_COND_MSG(p_peer.is_valid() && p_peer->get_connection_status() == NetworkedMultiplayerPeer::CONNECTION_DISCONNECTED,
```

Comment:
IINM one intended use for this method could be to reset the network peer to null with `set_network_peer(NULL)`.
I think it's still going to work as `p_peer.is_valid()` here will fail, so it won't error out, and `network_peer = p_peer;` will still run as expected.